locator/topology.cc

```diff
@@ -341,7 +341,7 @@ void topology::index_node(const node* node) {
                 on_internal_error(tlogger, format("topology[{}]: {}: node endpoint already mapped to {}", fmt::ptr(this), node_printer(node), node_printer(eit->second)));
             }
         }
-        if (node->get_state() != node::state::left) {
+        if (!node->left() && !node->is_none()) {
```

Comment:
@bhalevy please review " locator: topology: add_or_update_endpoint: use none as the default node state" commit

Reply:
We do not test gossiper mode any longer AFAIK, but we do changes to it. This is scary.

Reply:
We have two pipelines for this reason

Reply:
I wonder why do you need an own new state? Can't you use the fact that the node has no tokens here instead?

Reply:
> Can't you use the fact that the node has no tokens here instead?

Different layer of abstraction. `locator::topology` is not aware of token ownership

> I wonder why do you need an own new state?

Can you rephrase WDYM? No new states are introduced here. (`node::state::none` is pre-existing, but it was used (or rather not used) incorrectly)